src/common/SmartUrlInput.js

```diff
@@ -96,22 +98,22 @@ export default class SmartUrlInput extends PureComponent<Props, State> {
     const {
       defaultOrganization,
       protocol,
-      append,
+      defaultHostname,
       defaultValue,
       style,
       onSubmitEditing,
       enablesReturnKeyAutomatically,
     } = this.props;
-    let { value } = this.state;
-    let showAppend = value.indexOf('.') === -1;
-    if (defaultValue && value.length === 0) {
-      showAppend = false;
-      value = defaultValue;
-    }
+    const { value } = this.state;
+
+    const [prefix, , suffix] = autocompleteHostnamePieces(value ?? defaultValue, {
```

Comment:
It is, though! I'm reluctant to condemn TC-39-blessed syntax just because it looks unusual at first. At [the top of that ESlint page](https://eslint.org/docs/rules/no-unused-vars), there's even an example using it.

Aesthetically, I also prefer `_`-as-ignored-value (as also seen in at least Rust and Haskell), but you'd still get complaints if you're ignoring more than one value. (Unless you name them... but then you have to name them, and then someone *will* eventually come along and blithely use those names.)

Also, while it's not in use in zulip-mobile, some people do bind `_` to lodash.

The argument for `const { prefix, suffix } = ...` is a lot stronger, of course (as anyone who's ever used [the perl `stat` function](https://perldoc.perl.org/functions/stat.html) can attest). I think returning a tuple is okay for this particular case, since it's chopping a string (which is inherently sequential) into a very specific and very small number of pieces. But it's marginal, at best.